packages/codec/README.md

```diff
@@ -1,15 +1,161 @@
-# Truffle Decoder Core
-This module provides interfaces for decoding Solidity variables.  This is a
-fairly low-level interface meant to be used by e.g. a debugger; for a
-higher-level interface, see the `truffle-contract-decoder` package instead.
+# Truffle Decoder
+
+This module provides interfaces for decoding contract state, transaction
+calldata, and events.
+
+It's split into three classes: The wire decoder, the contract decoder, and the
+contract instance decoder.  The wire decoder is associated to the project as a
+whole and decodes transaction calldata and events.  The contract decoder is
+associated to a specific contract class.  It has all the capabilities of the
+wire decoder, but in addition it acts as a factory for contract instance
+decoders.  The contract instance decoder is associated to a specific contract
+instance; it too has all the capabilities of the wire decoder, but it can also
+decode the state variables for the specific instance.  (In addition, in the case
+that the contract does not include a `deployedBytecode` field in its artifact,
+which can hinder decoding certain things, the contract instance decoder can
+sometimes work around this where the other decoders cannot.)
+
+This documentation describes the current state of the decoder, but you should
+expect to see improvements soon.  Note that most of the documentation is not
+found in this README, but rather in this package's TypeDoc.  However, this
```

Comment:
> (with "API documentation" being a link)

How would I do that?  Wouldn't we need to know the URL in advance?

> Also, TypeDoc renders README.md by default, so people will probably read this document for the first time while already looking at the TypeDoc. It might be worth keeping a separate TypeDoc README

How would that work?

Reply:
Re: link inside repo README: you're right, let's solve this one later.

Re: separate README: see TypeDoc docs. It's probably a config/CLI option.

Reply:
(captured this as note for follow-up issue)